### PR TITLE
Add dry-core gem to site

### PR DESCRIPTION
**Pull request by davydovanton**
/cc @flash-gordon @AMHOL

### CODE REVIEW

File: source/gems/dry-core/index.html.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: Introduction &amp; Usage
+description: A toolset of small support modules used throughout the @dry-rb & @rom-rb ecosystems
+layout: gem-single
+order: 5
+type: gem
+name: dry-core
+---
+
+`dry-core` is a simple toolset that can be used in any places.
```
Comment (AMHOL) [inline source/gems/dry-core/index.html.md]: Should this be `many places`?

File: source/gems/dry-core/index.html.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Usage
+
+### Cache
+llows you to cache call results that are solely determined by arguments.
```
Comment (AMHOL) [inline source/gems/dry-core/index.html.md]: Typo `llows` => `Allows`

File: source/gems/dry-core/index.html.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  def self.old_class_api; end
+  def self.new_class_api; end
+
+  deprecate_class_method :old_class_api, :new_class_api
+
+  def old_api; end
+  def new_api; end
+
+  deprecate_method :old_api, :new_api, "old_api is no-no"
```
Comment (AMHOL) [inline source/gems/dry-core/index.html.md]: Perhaps we could call these and show the deprecation message in comments?

**Comment by AMHOL** [comment]
Thanks @davydovanton, couple of typos but looks good otherwise

**Comment by davydovanton** [comment]
@AMHOL thanks for the review! 👏 
I fixed typos, and also I added working code as an example for `Deprecations` module

File: source/gems/dry-core/index.html.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+* `EMPTY_ARRAY`
+* `EMPTY_HASH`
+* `EMPTY_OPTS`
+* `EMPTY_SET`
+* `EMPTY_STRING`
+* `Undefined` - A special value you can use as a default to know if no arguments were passed to you method
+
+### Deprecations
+
+Fox deprecate ruby methods you need to extend `Dry::Core::Deprecations` module
```
Comment (AMHOL) [inline source/gems/dry-core/index.html.md]: Not sure what `Fox deprecate ruby methods` means here? Should it be `For deprecating ruby methods, ...`?

Comment (davydovanton) [inline source/gems/dry-core/index.html.md]: sorry, fixed too!

**Comment by flash-gordon** [comment]
awesome, @davydovanton, thanks for taking care of this :)

**Pull request merged**